src/jit/compiler.h

```diff
@@ -3107,7 +3107,7 @@ class Compiler
     bool isScalarISA(InstructionSet isa);
     static int ivalOfHWIntrinsic(NamedIntrinsic intrinsic);
     unsigned simdSizeOfHWIntrinsic(NamedIntrinsic intrinsic, CORINFO_SIG_INFO* sig);
-    static int numArgsOfHWIntrinsic(NamedIntrinsic intrinsic);
+    static int numArgsOfHWIntrinsic(NamedIntrinsic intrinsic, GenTreeHWIntrinsic* node);
```

Comment:
You haven't set the default parameter value; however, it seems that all callers are passing in the node.
Either the function header comment needs to be fixed, or the default value should be set here.

Reply:
Oh my. It is my local merge error - it was set

Reply:
Are there any scenarios where we will be looking this up without having a corresponding node?

Reply:
If there are not, I would vote not having any default and actually asserting that `node != null`, since we generally want to be ensuring that the lookup returns the correct value.

Reply:
I don’t think this function should have nullptr as the default value of node. Every call site does not know the intrinsic is variable args or not. One-arg calling will lead misusing.

Reply:
Yes, looking at the call sites, we don't use this method in a place where prior knowledge of the intrinsic would be available. So, I agree - no default is the way to go. Sorry for the distraction.